packages/frontend/src/components/cart/Cart.tsx

```diff
@@ -110,16 +113,15 @@ export default function Cart({
   async function handleCheckout() {
     try {
       await onCheckout!();
-    } catch (error) {
+    } catch (error: unknown) {
       if (
-        error instanceof Error && (
-          error.message === "not enough items in stock for order" ||
-          error.message == "not enough stock" ||
-          error.message == "not in stock"
-        )
+        // TODO: This is a big turd but i couldn't make it recdognize RelayResponseError for the life of me.
+        error.originalError.code === 9 && error.originalError.additionalInfo
       ) {
-        setErrorMsg("Not enough stock. Cart cleared.");
-        await clearCart();
+        const objectId = error.originalError.additionalInfo.object_id;
+        console.log("Object ID:", objectId);
+        setErrorMsg(`Item ${objectId} is out of stock.`);
+        // await clearCart();
```

Comment:
this is just stapled together, too. should make a nicer toaster message using the item title, etc.